### PR TITLE
Add release process documentation

### PR DESCRIPTION
**Pull request by chmouel**
/cc @vdemeester @hrishin @sthaha @danielhelfand

### CODE REVIEW

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+- When it's released you can build the rpm according to this
+  [README](tekton/rpmbuild/README.md). Make sure you have done the PREREQ to be
+  able to upload it to the copr repository. This could take some time if copr
+  builder is busy.
+
+- you need to edit the Changelog according to the other templates in there,
+  usually you may want to do this in hackmd so you can have other cli
+  maintainers do this with you. an example
+  [here](https://hackmd.io/Gj9bxTh0Qn6kOaEx8gjsJg?edit).
```
Comment (piyush-garg) [inline RELEASE_PROCESS.md]: Try updating the url to https://hackmd.io/@chmouel/SykvrTGoB/

Comment (chmouel) [reply RELEASE_PROCESS.md]: lol i made a edit and pushed --force but forgot to save in my editor!

File: RELEASE_PROCESS.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# TektonCD CLI release process
+
+- You need to be a tektoncd-cli [OWNERS](OWNERS) to be able to push the new branch
+
+- You need to have your own tekton cluster, under `minikube`, `gcp` or other means.
```
Comment (vdemeester) [inline RELEASE_PROCESS.md]: nit: at some point, we want to automate release on the `dogfooding` cluster :wink: but we are not yet there for sure :sweat_smile:

File: RELEASE_PROCESS.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+```
+
+- Make an update for [homebrew-core](https://github.com/Homebrew/homebrew-core/blob/master/Formula/tektoncd-cli.rb) for tektoncd-cli formula, make sure you get the proper `sha256sum` and update it like this :
+
+  https://github.com/Homebrew/homebrew-core/pull/46492
+
+- Announce and spread the love on twitter, make sure you tag
+  [@tektoncd](https://twitter.com/tektoncd) account so you get retweeted and
+  perhaps add the major new feature in the tweet. See [here](https://twitter.com/chmouel/status/1177172542144036869) for an example.
+  Do not fear to add a bunch of  emojis 🎉🥳. Ask @vdmeester for tips 🤣.
```
Comment (vdemeester) [inline RELEASE_PROCESS.md]: :heart:

File: RELEASE_PROCESS.md (hunk 2 above)
Comment (chmouel) [reply RELEASE_PROCESS.md]: :)

**Pull request merged**